testsuite/tests/contrib/lkt_semantic/builtin_methods/test.out

```diff
@@ -1,206 +1,169 @@
 Resolving test.lkt
 ==================
-Id   <RefId "Node" test.lkt:1:14-1:18>
-     references <ClassDecl "Node" __prelude:27:10-30:2>
+Id   <RefId "Node" test.lkt:1:33-1:37>
+     references <GenericDecl "Node" __prelude:27:10-30:2>
 
-Id   <RefId "TokenNode" test.lkt:1:30-1:39>
-     references <TraitDecl "TokenNode" __prelude:31:10-33:2>
+Id   <RefId "Name" test.lkt:1:38-1:42>
+     references <ClassDecl "Name" test.lkt:1:11-11:2>
 
 Id   <RefId "Node" test.lkt:3:25-3:29>
-     references <ClassDecl "Node" __prelude:27:10-30:2>
+     references <TraitDecl "Node" __prelude:27:36-30:2>
 
 Id   <RefId "node" test.lkt:3:32-3:36>
-     references <NodeDecl "node" test.lkt:1:1-11:2>
+     references <NodeDecl "node" test.lkt:1:11-11:2>
 
 Expr <RefId "node" test.lkt:3:32-3:36>
-     has type <ClassDecl "Name" test.lkt:1:1-11:2>
+     has type <ClassDecl "Name" test.lkt:1:11-11:2>
 
 Id   <RefId "parent" test.lkt:3:37-3:43>
-     references <FunDecl "parent" __prelude:28:24-28:42>
+     references <FunDecl "parent" __prelude:28:24-28:54>
 
 Expr <RefId "parent" test.lkt:3:37-3:43>
-     has type <ClassDecl "Node" __prelude:27:10-30:2>
+     has type <ClassDecl "Name" test.lkt:1:11-11:2>
 
-Expr <DotExpr test.lkt:3:32-3:43>
-     has type <ClassDecl "Node" __prelude:27:10-30:2>
+test.lkt:3:32: error: Mismatched types: expected `Node`, got `Name`
+2 |     fun parent_node (): Node = node.parent
+  |                                ^^^^^^^^^^^
+
+test.lkt:5:24: error: Invalid type reference
+4 |     fun parent_env (): LexicalEnv = node.parent.node_env()
+  |                        ^^^^^^^^^^                         
```

Comment:
Looks like the testcase needs more updates :-)

Reply:
Good catch!

Reply:
Yup, looks like you're going to have to modify `dsl_unparse.py` again :)

Here apparently you're going to need to replace `LexicalEnv` by `LexicalEnv[<name of root node type>]` everywhere.

I think that eventually we want to think of a shortcut way to refer to those things, like we discussed yesterday with @Roldak, but we need to think about this a little bit more, so in the meantime let's update the tests with the verbose syntax.

Reply:
Done, thanks for the help!